src/messaging/messaging.ts

```diff
@@ -228,6 +235,55 @@ export class Messaging implements MessagingInterface {
     return this.appInternal;
   }
 
+  /**
+   * Converts from the public {@link messaging.Message `Message`} type to the
+   * internal/generated {@link protos.google.firebase.fcm.v1.Message `clientMessage`}
+   * type.
+   *
+   * @param message The {@link messaging.Message `Message`} to be converted
+   * @returns A {@link protos.google.firebase.fcm.v1.Message `clientMessage`},
+   * where like fields are the same as the inptuted message.
+   */
+  private convertToClientMessage(message: Message): ClientMessage {
+    //TODO: Add conversion for android, webpush, apns
+
+    const clientMessage: ClientMessage = {
```

Comment:
Can we just do this:

```ts
return {
  data: message.data,
  notification: message.notification,
  fcmOptions: message.fcmOptions,
  token: message.token,
  topic: message.topic,
  condition: message.condition,
}
```

Reply:
Sorry if this has already been noted, but shouldn't we have converters for each type?
```
return {
  data: message.data,
  notification: convertToClientNotification(message.notification),
  fcmOptions: convertToFcmOptions(message.fcmOptions),
  token: message.token,
  topic: message.topic,
  condition: message.condition,
}
``
Otherwise we could just `return message as ClientMessage`?

Reply:
data, notification, and fcmOptions are easily converted because they only contain primitive types. In the future, there will be converters for the types that have more complicated structuring (i.e `androidConfig` and `webPush`). 
The suggestion of just returning an object with `token`, `topic` and `condition` regardless of the type of the `message` parameter would be cleaner, but does not compile.